pacman/game.py

```diff
@@ -51,19 +52,31 @@ def initialize_pac_dots(self):
         for node_pos in self.board.nodes:
             self.pac_dots[node_pos] = self.pac_dot_reward
 
+    def add(self, name, position):
+        name = name.lower()
+        init_node = self.board.nodes[position]
```

Comment:
All the error handling does not work here because if position is not in self.board.nodes, we get a Key error and all the rest is not exectuted, the InvalidPosition error is therefore never raised.

We can maybe remove the original error handling and just rely on the native KeyError ? Or check this error case here specifically and raise a more explicit error.